File: src/main/java/net/serebryansky/common/service/PlaceService.java
package net.serebryansky.common.service;

import net.serebryansky.place.model.Place;
import org.reactivestreams.Publisher;
import reactor.core.publisher.Flux;
import reactor.core.publisher.Mono;

public interface PlaceService {
    Flux<Place> getPlaces();

    Mono<Place> getPlace(String id);
}

File: src/main/kotlin/net/serebryansky/common/service/PlaceService.kt
package net.serebryansky.common.service

import kotlinx.coroutines.flow.Flow
import net.serebryansky.place.model.Place
import reactor.core.publisher.Flux
import reactor.core.publisher.Mono

interface PlaceService {
    suspend fun getPlaces(): Flow<Place>
    suspend fun getPlace(id: String): Place
}
